### PR TITLE
Create program for ADADAS dataset

### PR DESCRIPTION
**Pull request by kaz462**
adadas initial draft, pending updates in define.xml

### CODE REVIEW

File: vignettes/adam-adadas.Rmd
Comment (SHAESEN2) [inline vignettes/adam-adadas.Rmd:1]: Why do we need an RMD file for ADaM creation?

File: vignettes/ectd/adam-adadas.R
Comment (SHAESEN2) [inline vignettes/ectd/adam-adadas.R:1]: name of file should be adadas.R, not adam-adadas.R.

File: vignettes/ectd/adam-adadas.R
Comment (SHAESEN2) [inline vignettes/ectd/adam-adadas.R:6]: We should agree on a standard workflow at the top of the program. Typically you dont need to initialize the startup.R and do a check for the writing permission.

I only have two lines:

# Set up ------------------------------------------------------------------

fcts <- c("eff_models.R", "fmt.R", "helpers.R", "Tplyr_helpers.R")
invisible(sapply(fcts, FUN = function(x) source(file.path("R/", x), )))

File: vignettes/ectd/adam-adadas.R
Comment (SHAESEN2) [inline vignettes/ectd/adam-adadas.R:133]: Missing:
1. QC versus the original xpt file to ensure you have 100% match
2. export dataset. I propose haven::write_xpt(adtte, file.path("submission/datasets/adtte.xpt")) to mimick an actual submission environment

**Review by SHAESEN2 — CHANGES_REQUESTED** [review]

File: vignettes/ectd/adam-adadas.R
Comment (SHAESEN2) [reply vignettes/ectd/adam-adadas.R:133]: Let me know if you need any help :)

File: vignettes/adam-adadas.Rmd
Comment (thomas-neitmann) [reply vignettes/adam-adadas.Rmd:1]: No, we discussed on Friday to use plain `.R` files.

File: vignettes/ectd/adam-adadas.R
Comment (thomas-neitmann) [reply vignettes/ectd/adam-adadas.R:6]: Agree, no need for sourcing the startup script and checking write permissions. I think we don't even need the two lines you suggested @SHAESEN2. Since the project is a R package and the functions are in the `R/` folder we just have to `library()` the pakage to make the function available.

File: vignettes/ectd/adam-adadas.R
Comment (thomas-neitmann) [reply vignettes/ectd/adam-adadas.R:133]: For 2. please use {xportr} which works in conjunction woth {metacore} and {metatools} to create a valis `.xpt` file.

Comment (thomas-neitmann) [reply vignettes/ectd/adam-adadas.R:133]: For 1. I would propose to put the comparing of datasets in a separate file `qc_adas.R` which reads in the original SAS dataset, the dataset you created and uses `diffdf::diffdf()` for comparison.

File: vignettes/adam-adadas.Rmd
Comment (thomas-neitmann) [inline vignettes/adam-adadas.Rmd:112]: A bit overkill but please use `derive_vars_duration()`.

**Review by thomas-neitmann — CHANGES_REQUESTED** [review]

File: vignettes/adam-adadas.Rmd
Comment (kaz462) [reply vignettes/adam-adadas.Rmd:112]: `derive_vars_duration()` is not applicable here:
* `derive_vars_duration()`  derives duration from **dates**, but AWTARGET is the target day within the visit window defined in SAP, not dates;
* AWTDIFF is the absolute difference between AWTARGET - ADY, not duration.

**Comment by kaz462** [comment]
Thanks @SHAESEN2 @thomas-neitmann for the review! I've addressed the above comments, also added program headers as @SHAESEN2 did in `submission/programs/adtte.R`. 

Please suggest on folder structures for `.R` and output xpt dataset. I put `adadas.R` and `qc_adadas.R `under vignettes/ectd/ for now as discussed in the last meeting.

File: vignettes/ectd/adam-adadas.R
Comment (SHAESEN2) [reply vignettes/ectd/adam-adadas.R]: > Thanks @SHAESEN2 @thomas-neitmann for the review! I've addressed the above comments, also added program headers as @SHAESEN2 did in `submission/programs/adtte.R`.
> 
> Please suggest on folder structures for `.R` and output xpt dataset. I put `adadas.R` and `qc_adadas.R `under vignettes/ectd/ for now as discussed in the last meeting.

@kaz462 I proposed a folder structure for the dataset and program in #21 #27 waiting for team approval. Not sure where the qc program should live. Perhaps a qc folder in root?

**Comment by SHAESEN2** [comment]
> Thanks @SHAESEN2 @thomas-neitmann for the review! I've addressed the above comments, also added program headers as @SHAESEN2 did in `submission/programs/adtte.R`.
> 
> Please suggest on folder structures for `.R` and output xpt dataset. I put `adadas.R` and `qc_adadas.R `under vignettes/ectd/ for now as discussed in the last meeting.

@kaz462 I proposed a folder structure for the dataset and program in #21 #27 waiting for team approval. Not sure where the qc program should live. Perhaps a qc folder in root?

File: vignettes/adam-adadas.Rmd
Comment (thomas-neitmann) [reply vignettes/adam-adadas.Rmd]: I see, never mind then 🙂

File: vignettes/ectd/qc_adadas.R
Comment (thomas-neitmann) [inline vignettes/ectd/qc_adadas.R:17]: Why is this commented? Also keys have to be defined for the `diffdf()` call.

File: vignettes/ectd/adadas.R
Comment (thomas-neitmann) [inline vignettes/ectd/adadas.R:22]: Please remove these kind of comment lines, they don't serve a purpose.

**Review by thomas-neitmann — CHANGES_REQUESTED** [review]

**Comment by thomas-neitmann** [comment]
I like the structure of `submission/datasets` and `submission/programs`. As the QC program will not be part of the submission a `qc` folder in the root directly makes sense. Please remember to add these non.standard directories (for an R package) to `.Rbuildignore`. Thanks!

File: renv.lock
Comment (thomas-neitmann) [inline renv.lock:1487]:
```suggestion
      "Repository": "CRAN",
```

**Comment by thomas-neitmann** [comment]
I'm going ahead and merge this even though one parameter has still discrepancies as I'd like to get some updates made on this branch onto `main`. @kaz462 Please open a new issue and PR for the final updates once you are back. Thanks!

**Pull request merged**
